src/legacy/core_plugins/data/public/expression_executor/expression_executor_service.tsx

```diff
@@ -0,0 +1,177 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup } from 'kibana/public';
+import { useRef, useEffect } from 'react';
+import React from 'react';
+
+import { fromExpression, Ast } from '@kbn/interpreter/common';
+
+// this type import and the types below them should be switched to the types of
+// the interpreter plugin itself once they are ready
+import { RunPipelineHandlers } from 'ui/visualize/loader/pipeline_helpers/run_pipeline';
+import { Registry } from '@kbn/interpreter/common';
+import { RequestAdapter, DataAdapter } from 'ui/inspector/adapters';
+
+type Context = object;
+type Handlers = RunPipelineHandlers;
+export interface Result {
+  type: string;
+  as?: string;
+  value?: unknown;
+}
+
+interface RenderHandlers {
+  done: () => void;
+  onDestroy: (fn: () => void) => void;
+}
+
+export interface RenderFunction {
+  name: string;
+  displayName: string;
+  help: string;
+  validate: () => void;
+  reuseDomNode: boolean;
+  render: (domNode: Element, data: unknown, handlers: RenderHandlers) => void;
+}
+
+export type RenderFunctionsRegistry = Registry<unknown, RenderFunction>;
+
+export interface Interpreter {
+  interpretAst(ast: Ast, context: Context, handlers: Handlers): Promise<Result>;
+}
+
+export interface ExpressionExecutorSetupPlugins {
+  interpreter: {
+    renderersRegistry: RenderFunctionsRegistry;
+    getInterpreter: () => Promise<{ interpreter: Interpreter }>;
+  };
+}
+
+async function runFn(
+  expressionOrAst: string | Ast,
+  element: Element,
+  renderersRegistry: RenderFunctionsRegistry,
+  interpreter: Interpreter
+) {
+  const ast =
+    typeof expressionOrAst === 'string' ? fromExpression(expressionOrAst) : expressionOrAst;
+  const response = await interpreter.interpretAst(
+    ast,
+    { type: 'null' },
+    {
+      getInitialContext: () => ({}),
+      inspectorAdapters: {
+        // TODO connect real adapters
+        requests: new RequestAdapter(),
+        data: new DataAdapter(),
+      },
+    }
+  );
+
+  if (response.type === 'render' && response.as) {
+    renderersRegistry.get(response.as).render(element, response.value, {
+      onDestroy: fn => {
+        // TODO implement
+      },
+      done: () => {
+        // TODO implement
+      },
+    });
+  } else {
+    // eslint-disable-next-line no-console
+    console.log('Unexpected result of expression', response);
+  }
+
+  return response;
+}
+
+/**
+ * Expression Executor Service
+ * @internal
+ */
+export class ExpressionExecutorService {
+  private interpreterInstance: Interpreter | null = null;
+
+  // TODO core won't ever be null once this is switched to the new platform
```

Comment:
This was a misunderstanding from my side, sorry. I thought the individual services were planned to become plugins in the future because of the very similar call signature. I removed the core parameter completely, this also should make clear the service isn't depending on it